tests/api_tests.js

```diff
@@ -331,89 +329,56 @@ function runNodeEnvTests(nodeEnvData, t) {
-			
-			// If we reach the end of the end of the loop, we passed and end
-		// });
-			
-	// };
-	
-	// test the same field in every function is ok
+function runProfilingTests(profData, t){
+	var functions = profData['functions']; 
 	
-	
-	
-// }
-
-
-
-function runProfilingTests(profData, t) {
-
-	var prefix = prefix || 'Profiling message ';
-
 	t.ok(isInteger(profData.time), "Timestamp is an integer");
 	t.ok(isReasonableTimestamp(parseInt(profData.time)), "Timestamp is a reasonable value (expected to contain current year)");
-
-	var functions = profData['functions'];
 	
-	for (var index in functions){
-		var currentFunction = functions[index];
-		
-		
-		if (!isInteger(currentFunction.self)){ 
-			t.fail(prefix + "does not have an integer id (" + currentFunction.self + ")");
-		}
-		
-		if (!isInteger(currentFunction.parent)){ 
-			t.fail(prefix + "does not have an integer parent id (" + currentFunction.parent + ")");
-		}
-		
-		if (!isInteger(currentFunction.line)){ 
-			t.fail(prefix + "does not have an integer line number(" + currentFunction.line + ")");
-		}
-
-		if (!isInteger(currentFunction.count)){
-			t.fail(isInteger(currentFunction.count),
-			   prefix + "does not have an integer sample count (" + currentFunction.count + ")");
-		}
-
-
-		for (var entry in currentFunction)
-		{
-		  if (entry != 'file' || entry != 'name')
+	testValuesAreIntegers("self");
+	testValuesAreIntegers("parent");
+	testValuesAreIntegers("line");
+	testValuesAreIntegers("count");
+	
+	//Parse values of all functions for next tests
+	for (var currentFunction in functions){
+		for (var entry in currentFunction){
+			if (entry != 'file' || entry != 'name')
 			currentFunction[entry] = parseInt(currentFunction[entry]);
 		}
-
-		if (currentFunction.self <= 0){ 
-			t.fail(prefix + "does not have a positive id (" + currentFunction.self + ")");
-		}
-		
-		if (currentFunction.parent < 0){ 
-			t.fail(prefix + "does not have a positive (or 0) parent id (" + currentFunction.parent + ")");
-		}
-		
-		if (currentFunction.line < 0){
-			t.fail(prefix + "does not have a positive (or 0) line number (" + currentFunction.line + ")");
+	}
+	
+	testValuesAreNotNegative("self");
+	testValuesAreNotNegative("parent");
+	testValuesAreNotNegative("line");
+	testValuesAreNotNegative("count");
+	
+	//Check the same key for all functions in data are integer
+	function testValuesAreIntegers(keyName){
+		for (var index in functions){
+			if(!isInteger(functions[index][keyName])){ 
+				t.fail("Value of "+keyName+" should be an integer ("+	functions[index][keyName]+")");
+				return;
+			}
 		}
-		
-		
-		if (currentFunction.count < 0){
-			t.fail(prefix + "does not have a positive (or 0) sample count (" + currentFunction.count + ")");
+		t.pass("Value of '" + keyName + "' is an integer for all functions");
+	}
+	
+	//Check the same key for all functions in data aren't negative
+	function testValuesAreNotNegative(keyName){ 
```

Comment:
If you change this to testValuesAreGreaterThan(keyName, val) then you can get rid of the special case for self and put the commend where you call testValuesAreGreaterThan('self', 1);